litmus/director/TCID-KUBELET-RESTART/test.yml

```diff
@@ -0,0 +1,71 @@
+---
+- hosts: localhost
+  connection: local
+
+  vars_files:
+    - test_vars.yml
+
+  tasks:
+    - block:
+
+          # Generating the testname for deployment
+        - include_tasks: /ansible-utils/create_testname.yml
+
+          # RECORD START-OF-TEST IN LITMUS RESULT CR
+        - include_tasks: /ansible-utils/update_litmus_result_resource.yml
+          vars:
+            status: 'SOT'
+
+         # Restart Kubelet based on platform
+        - block:
+
+            # Restart Kubelet container present on the node
+          - name: Restarting Kubelet container in Rancher
+            shell: docker restart kubelet
+
```

Comment:
To do this, we need root access right?

Reply:
@gprasath it will restart the docker on the node in which the pod is scheduled. We have made use of `Docker out of Docker` for this. We have mounted the host's docker socket to this pod for this.

Reply:
@gprasath what will happen to the statefulsets scheduled on that node if we stop the kubelet till node is in NotReady state?

Reply:
Yes we need root access, so we are running the container with `securityContext` -- `privileged: true`

Reply:
> @gprasath what will happen to the statefulsets scheduled on that node if we stop the kubelet till node is in NotReady state?

That statefulset replica pod will be in pending state if the node is not ready

Reply:
> @gprasath it will restart the docker on the node in which the pod is scheduled. We have made use of Docker out of Docker for this. We have mounted the host's docker socket to this pod for this.

Restart is instantaneous. There won't be any considerable impact.

Reply:
@gprasath I think I was not clear -- the above command will restart kubelet container running on the host node itself.